scripts/task_process/cache_status.py

```diff
@@ -0,0 +1,222 @@
+            node = node_map[event['Cluster'], event['Proc']]
+            nodes[node]['ResidentSetSize'][-1] = int(event['ResidentSetSize'])
+            if nodes[node]['StartTimes']:
+                nodes[node]['WallDurations'][-1] = eventtime - nodes[node]['StartTimes'][-1]
+            insertCpu(event, nodes[node])
+        elif event["MyType"] == "JobDisconnectedEvent" or event["MyType"] == "JobReconnectedEvent":
+            # These events don't really affect the node status
+            pass
+        else:
+            logging.warning("Unknown event type: %s" % event['MyType'])
+
+    logging.debug("There were %d events in the job log." % count)
+    now = time.time()
+    for node, info in nodes.items():
+        last_start = now
+        if info['StartTimes']:
+            last_start = info['StartTimes'][-1]
+        while len(info['WallDurations']) < len(info['SiteHistory']):
+            info['WallDurations'].append(now - last_start)
+        while len(info['WallDurations']) > len(info['SiteHistory']):
+            info['SiteHistory'].append("Unknown")
+
+def parseErrorReport(fp, nodes):
+    def last(joberrors):
+        return joberrors[max(joberrors, key=int)]
+    data = json.load(fp)
+    #iterate over the jobs and set the error dict for those which are failed
+    for jobid, statedict in nodes.iteritems():
+        if 'State' in statedict and statedict['State'] == 'failed' and jobid in data:
+            statedict['Error'] = last(data[jobid]) #data[jobid] contains all retries. take the last one
+
+# --- New code ----
+
+def storeNodesInfoInFile():
+    filename = "task_process/status_cache.txt"
+    # Open cache file and get the location until which the jobs_log was parsed last time
+    try:
+        if os.path.exists(filename) and os.stat(filename).st_size > 0:
+            logging.debug("cache file found, opening and reading")
+            nodes_storage = open(filename, "r")
+
+            last_read_until = int(nodes_storage.readline())
+            nodes = ast.literal_eval(nodes_storage.readline())
+            node_map = ast.literal_eval(nodes_storage.readline())
+            nodes_storage.close()
+        else:
+            logging.debug("cache file not found, creating")
+            last_read_until = 0
+            nodes = {}
+            node_map = {}
+    except Exception:
+        logging.exception("error during status_cache handling")
+    jobsLog = open("job_log", "r")
+
+    try:
+        errorSummary = open("error_summary.json", "r")
+        errorSummary.seek(0)
+        parseErrorReport(errorSummary, nodes)
+        errorSummary.close()
+    except IOError:
+        logging.exception("error during error_summary file handling")
+
+    jobsLog.seek(last_read_until)
+
+    parseJobLog(jobsLog, nodes, node_map)
+
+    read_until = jobsLog.tell()
+    jobsLog.close()
+    nodes_storage = open(filename, "w")
+    nodes_storage.seek(0)
+
+    nodes_storage.write(str(read_until) + "\n")
```

Comment:
We should write to a temporary file (like `'filename.%s' % os.getpid()`), and then move the file to the destination one (filename). Otherwise other process that reads the file (namely httpd) are screwed.